frontend/app/routes/_public+/apply+/_index.tsx

```diff
@@ -51,12 +53,28 @@ export async function action({ request }: ActionFunctionArgs) {
 
 export default function ApplyIndex() {
   const { siteKey } = useLoaderData<typeof loader>();
+  const captchaRef = useRef<HCaptcha>(null);
+  const submit = useSubmit();
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const formData = new FormData(event.currentTarget);
+
+    const executeResponse = captchaRef.current?.execute({ async: true });
+    executeResponse?.then((result) => {
+      formData.set('h-captcha-response', result.response);
+      submit(formData, { method: 'POST' });
+
+      captchaRef.current?.resetCaptcha();
+    });
+  }
```

Comment:
Code refactoring

- Set `handleSubmit` as `async` function then use `await` when executing `.execute({ async: true })`.

```suggestion
  async function handleSubmit(event: FormEvent<HTMLFormElement>) {
    event.preventDefault();
    if (captchaRef.current) {
      const { response } = await captchaRef.current.execute({ async: true });

      const formData = new FormData(event.currentTarget);
      formData.set('h-captcha-response', response);
      submit(formData, { method: 'POST' });

      captchaRef.current.resetCaptcha();
    }
  }
```

Reply:
/fixed